tests/units/txBatch/bulker.optimisation.test.ts

```diff
@@ -0,0 +1,303 @@
+      }
+
+      break;
+    }
+    case TransactionType.supply: {
+      mock.userMarketsData[Underlying.weth].walletBalance = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.weth].scaledSupplyOnPool = DELTA;
+      }
+      break;
+    }
+    case TransactionType.supplyCollateral: {
+      mock.userMarketsData[Underlying.dai].walletBalance = reverse
+        ? BASE_AMOUNT
+        : BASE_AMOUNT.mul(2);
+
+      if (reverse && relation === OperationRelation.bigger) {
+        mock.userMarketsData[Underlying.dai].scaledCollateral = DELTA;
+      }
+      break;
+    }
+  }
+  return mock;
+};
+
+describe("Bulker should merge", () => {
+  const userAddress = Wallet.createRandom().address;
+  let bulkerHandler: BulkerTxHandler;
+  let adapter: MorphoAaveV3Adapter;
+
+  afterEach(() => {
+    bulkerHandler.close();
+  });
+
+  [
+    TransactionType.supply,
+    TransactionType.borrow,
+    TransactionType.supplyCollateral,
+  ].forEach((baseTxSide) => {
+    [false, true].forEach((baseReverse) => {
+      [true, false].forEach((baseMax) => {
+        describe(`a ${getTxType(baseTxSide, baseReverse)}${
+          baseMax ? " max" : ""
+        } operation`, () => {
+          const baseTxType = getTxType(baseTxSide, baseReverse);
+          const baseOperation = {
+            type: baseTxType,
+            underlyingAddress:
+              baseTxSide === TransactionType.supplyCollateral
+                ? Underlying.dai
+                : Underlying.weth,
+            amount: baseMax ? constants.MaxUint256 : BASE_AMOUNT,
+          };
+
+          [
+            {
+              txType: baseTxType,
+              reverse: false,
+              max: false,
+            },
+            {
+              txType: baseTxType,
+              reverse: false,
+              max: true,
+            },
+            {
+              txType: baseTxType,
+              reverse: true,
+              max: false,
+              relation: OperationRelation.smaller,
+            },
+            {
+              txType: baseTxType,
+              reverse: true,
+              max: false,
+              relation: OperationRelation.equivalent,
+            },
+            {
+              txType: baseTxType,
+              reverse: true,
+              max: false,
+              relation: OperationRelation.bigger,
+            },
+            {
+              txType: baseTxType,
+              reverse: true,
+              max: true,
+              relation: OperationRelation.equivalent,
+            },
+            {
+              txType: baseTxType,
+              reverse: true,
+              max: true,
+              relation: OperationRelation.bigger,
+            },
+          ].forEach(({ txType, reverse, max, relation }) => {
+            it(`with a ${relation ? relation + " " : ""}${getTxType(
```

Comment:
you can leverage on a `describe.each` for a better formatting of the tests

Reply:
As far as I remember I tried but i had some limitations